src/cli/index.ts

```diff
@@ -152,25 +160,34 @@ async function samplesFromDirectory(dataDir: string, topLevelRefs: string[] | un
         let jsonSamples: StringInput[] = [];
         const schemaSources: SchemaTypeSource[] = [];
         const graphQLSources: GraphQLTypeSource[] = [];
+        const typeScriptSources: TypeScriptTypeSource[] = [];
+
         for (const source of await readFilesOrURLsInDirectory(dir)) {
             // FIXME: We do a type switch here, but we know which types we're putting in
             // in the function above.  It should separate it right away.
             if (isJSONSource(source)) {
                 jsonSamples = jsonSamples.concat(source.samples);
             } else if (isSchemaSource(source)) {
                 schemaSources.push(source);
-            } else {
+            } else if (isGraphQLSource(source)) {
                 graphQLSources.push(source);
+            } else if (isTypeScriptSource(source)) {
+                typeScriptSources.push(source);
             }
         }
-        if (jsonSamples.length > 0 && schemaSources.length + graphQLSources.length > 0) {
-            return panic("Cannot mix JSON samples with JSON Schema or GraphQL in input subdirectory");
+
+        if (jsonSamples.length > 0 && schemaSources.length + graphQLSources.length + typeScriptSources.length > 0) {
+            return panic("Cannot mix JSON samples with JSON Schems, GraphQL, or TypeScript in input subdirectory");
         }
-        if (schemaSources.length > 0 && graphQLSources.length > 0) {
-            return panic("Cannot mix JSON Schema with GraphQL in an input subdirectory");
+
+        const oneUnlessEmpty = <T>(xs: T[]) => (xs.length === 0 ? 0 : 1);
+        if (oneUnlessEmpty(schemaSources) + oneUnlessEmpty(graphQLSources) + oneUnlessEmpty(typeScriptSources) > 1) {
```

Comment:
Actually, this can just be written as `schemaSources.length + graphQLSources.length + typeScriptSource.length > 1`

Reply:
So we always only have one of these sources?

Reply:
Oh, sorry, ignore that comment.